### PR TITLE
Fix some sharing specs

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: spec/ruby/core/hash/shared/store.rb [hunk 1]
```diff
@@ -22,15 +22,15 @@ def key.reverse() "bar" end
     k1.should_receive(:hash).and_return(0)
     k2.should_receive(:hash).and_return(0)
 
-    h[k1] = 1
-    h[k2] = 2
+    h.send @method, k1, 1
+    h.send @method, k2, 2
```
Comment (jemc) [inline spec/ruby/core/hash/shared/store.rb]: Can you use parentheses here to be consistent with the other specs in this file?

File: spec/ruby/core/hash/shared/store.rb [hunk 2]
```diff
@@ -22,15 +22,15 @@ def key.reverse() "bar" end
...
     h.size.should == 2
   end
 
   it "accepts keys with private #hash method" do
     key = HashSpecs::KeyWithPrivateHash.new
     h = new_hash
-    h[key] = "foo"
+    h.send @method, key, "foo"
```
Comment (jemc) [inline spec/ruby/core/hash/shared/store.rb]: Same as above.

File: spec/ruby/core/hash/shared/store.rb (hunk 1 above)
Comment (kachick) [inline spec/ruby/core/hash/shared/store.rb]: Sure.

**Comment by jemc** [comment]
Thanks!

**Comment by kachick** [comment]
@jemc Thanks to merge and your review! :)

**Pull request merged**